### PR TITLE
Add WorldTileProvider to tshock config

### PR DESCRIPTION
**Pull request by SignatureBeef**
This allows the tshock tile providers to be switched via the config instead of cli args via tsapi, for environments where the command line args cannot be changed.

### CODE REVIEW

File: TShockAPI/TShock.cs [hunk 1]
```diff
@@ -383,6 +383,19 @@ public override void Initialize()
 				if (Config.Settings.EnableGeoIP && File.Exists(geoippath))
 					Geo = new GeoIPCountry(geoippath);
 
+				// check if a custom tile provider is to be used
+				switch(Config.Settings.WorldTileProvider)
```
Comment (csmir) [inline TShockAPI/TShock.cs]: Is it worth considering to call `.ToLower()` here? This gives the administrator some room for error in capitalization.

Comment (sgkoishi) [reply TShockAPI/TShock.cs]: Or maybe create an enum for it and let Newtonsoft.Json handle the input string?

Comment (SignatureBeef) [reply TShockAPI/TShock.cs]: both are valid, ill roll with `.ToLower()` for now since it wont need extra types to be defined.

**Review by sgkoishi — DISMISSED** [review]

**Review by csmir — DISMISSED** [review]
Lgtm

**Review by hakusaro — DISMISSED** [review]

**Pull request merged**